Tests/TuistDependenciesTests/DependenciesControllerTests.swift

```diff
@@ -190,11 +199,185 @@ final class DependenciesControllerTests: TuistUnitTestCase {
 
         // Then
         XCTAssertFalse(carthageInteractor.invokedFetch)
+        XCTAssertFalse(carthageInteractor.invokedUpdate)
+        XCTAssertTrue(carthageInteractor.invokedClean)
+
+        XCTAssertFalse(swiftPackageManagerInteractor.invokedFetch)
+        XCTAssertFalse(swiftPackageManagerInteractor.invokedUpdate)
+        XCTAssertTrue(swiftPackageManagerInteractor.invokedClean)
+
+        XCTAssertFalse(cocoaPodsInteractor.invokedFetch)
+        XCTAssertFalse(cocoaPodsInteractor.invokedUpdate)
+    }
+
+    // MARK: - Update
+
+    func test_update_carthage() throws {
+        // Given
+        let rootPath = try temporaryPath()
+        let dependenciesDirectoryPath = rootPath
+            .appending(component: Constants.tuistDirectoryName)
+            .appending(component: Constants.DependenciesDirectory.name)
+
+        let platforms = Set<Platform>([.iOS])
+        let carthageDependencies = CarthageDependencies(
+            [
+                .github(path: "Moya", requirement: .exact("1.1.1")),
+                .github(path: "RxSwift", requirement: .exact("2.0.0")),
+            ],
+            options: [.useXCFrameworks, .noUseBinaries]
+        )
+        let dependencies = Dependencies(
+            carthage: carthageDependencies,
+            swiftPackageManager: nil,
+            platforms: platforms
+        )
+
+        // When
+        try subject.update(at: rootPath, dependencies: dependencies)
+
+        // Then
+        XCTAssertFalse(carthageInteractor.invokedClean)
+        XCTAssertFalse(carthageInteractor.invokedFetch)
+        XCTAssertTrue(carthageInteractor.invokedUpdate)
+        XCTAssertEqual(carthageInteractor.invokedUpdateParameters?.dependenciesDirectory, dependenciesDirectoryPath)
+        XCTAssertEqual(carthageInteractor.invokedUpdateParameters?.dependencies, carthageDependencies)
+        XCTAssertEqual(carthageInteractor.invokedUpdateParameters?.platforms, platforms)
+
+        XCTAssertTrue(swiftPackageManagerInteractor.invokedClean)
+        XCTAssertFalse(swiftPackageManagerInteractor.invokedUpdate)
+        XCTAssertFalse(swiftPackageManagerInteractor.invokedFetch)
+
+        XCTAssertFalse(cocoaPodsInteractor.invokedFetch)
+        XCTAssertFalse(cocoaPodsInteractor.invokedUpdate)
+    }
+
+    func test_update_swiftPackageManger() throws {
+        // Given
+        let rootPath = try temporaryPath()
+        let dependenciesDirectoryPath = rootPath
+            .appending(component: Constants.tuistDirectoryName)
+            .appending(component: Constants.DependenciesDirectory.name)
+
+        let platforms = Set<Platform>([.iOS])
```

Comment:
as mentioned, I'd also rewrite this to have the type before the assignment (it's in more places, so just do search&replace)

Reply:
Maybe `swiftformat` can do some of this? I took a quick peak but couldn't find anything except this: https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#typeSugar